core/ordersync/ordersync.go

```diff
@@ -340,6 +346,39 @@ func (s *Service) GetOrders(ctx context.Context, minPeers int) error {
 	return nil
 }
 
+// PeriodicallyGetOrders periodically calls GetOrders. It waits a minimum of
+// approxDelay (with some random jitter) between each call. It will block until
+// there is a critical error or the given context is canceled.
+func (s *Service) PeriodicallyGetOrders(ctx context.Context, minPeers int, approxDelay time.Duration) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+
+		if err := s.GetOrders(ctx, minPeers); err != nil {
+			return err
+		}
+
+		// Note(albrow): The random jitter here helps smooth out the frequency of ordersync
```

Comment:
Did we write tests in Mesh Manager to verify this hypothesis, or is this just speculation? I could definitely see this being the case; however, I also think it's possible that different jitter values would improve this even more, and I also think it's possible that in some circumstances a jitter of 0 would be best.

Reply:
This idea is based on one simple assumption: when we re-deploy or update Mesh nodes we tend to do them all at once. This is not just the case for our infra; it's probably true for any other team that runs more than one node. If each of those nodes always used the same exact interval, it would lead to spiky behavior where the network as a whole sees a lot of ordersync requests coming in at the same time. Adding some random jitter makes this a lot less likely.

We can't really measure it in Mesh Manager right now because Mesh Manager tests only run for 15 minutes and the new interval is set to 1 hour. That said, the assumption seems really straightforward to me and at the very least adding a random jitter is unlikely to cause any harm.

Reply:
I thought that was the assumption that you were using. I can see this being true of the network today, but I suspect that it will not always be the case.

I didn't think that it was likely to cause harm either, hence the "non-blocking" preface.

Reply:
I thought that was probably the assumption that you were using. Thanks for clarifying that for me.